### DISCUSSION TITLE
${tags} not populated in PDF

### DISCUSSION THREAD

**Discussion by Wet-Bandit** [nextgenhealthcare/connect#4713] [Q&A] [answered]
I have a Document Writer destination with HTML to generate a PDF.
The PDF is created but none of the velocity variables are populated 
with my values that are in the connectorMap.  Am I missing something 
in the HTML that makes this happen?

I have another channel I wrote a year ago and it works fine.  I don't know what the problem is with this one.

Narupley available?
Mirth Connect Server 3.10.0
Built on November 5, 2020
Java version: 1.8.0_181

**Comment by tonygermano** [comment, marked as answer]
Normally that only happens if the velocity variable you are trying to use is not defined in the maps at all or it is defined and contains a null value.

Maybe refer back to the forum post from the last time you were working on this? https://forums.mirthproject.io/forum/mirth-connect/support/177478-html-template-help

Your final comment indicates there is likely an issue with your template.
```
Final reply.... Be careful with tools that generate the html for you, they mess up the token
names ${abc} by sticking tags inside like this... ${abc <span <ddd></ddd> </span>}, I had to
go through and check these tokens 3 times and finally found them all.
```

**Reply by Wet-Bandit** [reply]
That's not the same problem, before it would not produce a PDF file.

**Reply by pacmano1** [reply]
Seems to work fine for me in a simple use case:
On a Document Writer destination add `connectorMap.put('test','stuff')` in a transformer
In the html template:
```
<html>
${test}
</html>
```
Results in a PDF with the word "stuff" in it.

Maybe post some code / screenshots?

**Reply by pacmano1** [reply]
LOL.  HTML beautified that is a 1253 lines long.  I am going to duck a deep dive on that and let someone else answer.

**Reply by Wet-Bandit** [reply]
All you have to do is run it and see that mirth does not plug in the provider_name value.
No need to read the HTML.

**Reply by pacmano1** [reply]
> All you have to do is run it and see that mirth does not plug in the provider_name value.
> No need to read the HTML.

Not quite - I trust it does not work as you say so that means diving into your html to figure out what is corrupting the substitution.

**Reply by Wet-Bandit** [reply]
Well, maybe, but it does work, I get a PDF file, but the PDF still shows ${provider_name}.

**Comment by Wet-Bandit** [comment]
RESOLVED:  The main bug was that there was a blank space in a few of the velocity variables.  (difficult to notice)
(I need to get the business analyst people to be more careful.)
